# simple-alignment
Needleman-Wunsch and Smith-Waterman Alignment Algorhithm as a Java Applet

This project was written by Connor Runyan as an exercise in applying his study of Computer Science to some basic bioinformatics algorithms.  Started Feb 8 2016.

This program uses the computationally intensive Needleman-Wunsch algorithm to globally align sequences and the equally intensive Smith-Waterman local alignment algorithm to compare sequences of values.

These algorithms were originally designed to compare sequenced DNA / RNA / Protein sequences, but are also suitable for aligning any two strings of characters.

This program will only compare a maximum of two strings of characters.  You can tweak the individual scoring schemes as well, if you want to further penalize gaps.
